### PR TITLE
Using ParseOp in TPTP

### PR DESCRIPTION
**Pull request by HanielB**
Makes the TPTP parser use ParseOp rather than a myriad of calls to the expression manager. This will help us in the parser migration.

This also adds a missing check in the SMT2 parsing relative to HOL.

### CODE REVIEW

File: src/parser/tptp/Tptp.g [hunk 1]
```diff
@@ -892,30 +1046,25 @@ thfLogicFormula[CVC4::Expr& expr]
...
+          case tptp::NA_REVAND:
+            expr = MK_EXPR(kind::NOT, MK_EXPR(kind::AND, expr, expr2));
+            break;
         }
       }
     | // N-ary and &
-      ( { args.push_back(expr); }
-        ( AND_TOK thfUnitaryFormula[expr] { args.push_back(expr); } )+
+      ( {
+  args.push_back(expr); }
```
Comment (ajreynol) [inline src/parser/tptp/Tptp.g]: Fix the clang-formatting here?

File: src/parser/tptp/Tptp.g [hunk 2]
```diff
@@ -883,7 +1034,10 @@ thfLogicFormula[CVC4::Expr& expr]
         {
           // TODO create whatever lambda
         }
-        expr = MK_EXPR(kind::EQUAL, expr, expr2);
+        args.push_back(expr);
+        args.push_back(expr2);
+        ParseOp p(kind::EQUAL);
+        expr = PARSER_STATE->applyParseOp(p, args);
```
Comment (ajreynol) [inline src/parser/tptp/Tptp.g]: Nit: this is a bit over-refactored, this line could have stayed the same.  (Same applies to a few other places).

Comment (HanielB) [reply src/parser/tptp/Tptp.g]: In `applyParseOp` there are checks for using HOL stuff. We'd have to move them here otherwise.

Comment (ajreynol) [reply src/parser/tptp/Tptp.g]: I see, sounds good.

**Pull request merged**